Employee_details_using_linkedlists/linkedlist.c

```diff
@@ -0,0 +1,290 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include "linkedlist.h"
+enum sort_or_remove{BY_NAME,BY_ID};
+enum menu{ADD=1,VIEW,SORT_BY_NAME,REMOVE_BY_ID,SORT_BY_ID,REMOVE_BY_NAME,EXIT};
+
+bool stringValidation(char *employee_name)
+{
+	int string_length=strlen(employee_name),indexOfchar=0;
+	if(string_length>NAME_LENGTH)//macros
+	{
+		printf("more than 10 characters!!\n");
+		return false;
+	}
+	while(indexOfchar<string_length)
+	{
+		if((employee_name[indexOfchar]>='A' && employee_name[indexOfchar]<='Z')||( employee_name[indexOfchar]>='a' && employee_name[indexOfchar]<='z'))
+		{
+			indexOfchar++;
+			continue;
+		}
+		else
+		{
+			return false;
+		}
+   }
+	return true;
+}
+struct employee* add_employee(struct employee* first_employee ) //remove this argument
+{
+	 //remove int details
+	 //remove this array
+	struct employee *employee_ptr,*temp_ptr;
+	if(!(employee_ptr=(struct employee *)malloc(sizeof(struct employee)))) //malloc validation
+	{
+		printf("memory not allocated\n");
+		return first_employee;
+	}
+
+	//getting employee name
+	 //change as discussed
+	do
+	{
+		printf("enter the name of the employee  :");
+		scanf("%s",employee_ptr->employee_name);
+	} while(!stringValidation(employee_ptr->employee_name));
+  //strcpy(employee_ptr->employee_name,employee_details);
+	// getting employee id
+	do
+	{
+		printf("enter the id of the employee  :");
+		scanf("%5d",&employee_ptr->employee_id); //integer validation required
+		if(getchar()!='\n')
+		{
+			while(getchar()!='\n');
+		}
+		else
+		{
+			break;
+		}
+	} while(1);
+
+	//getting employee age
+	do
+	{
+		printf("enter the age of the employee  :");
+		scanf("%2d",&employee_ptr->employee_age); //integer validation required
+		if(getchar()!='\n')
+		{
+			while(getchar()!='\n');
+		}
+		else
+		{
+			break;
+		}
+	} while(1);
+
+	//getting employee designation
+	do
+	{
+		printf("enter the designation of the employee  :");
+		scanf("%s",employee_ptr->employee_designation);
+	} while(!stringValidation(employee_ptr->employee_designation));
+
+	if(!first_employee)
+	{
+		employee_ptr->next_employee=NULL;
+		first_employee=employee_ptr;
+	}
+	else
+	{
+		printf("2nd node\n");
```

Comment:
remove all debug prints

Reply:
Yes I removed  everything in the recent commit.